### ISSUE TITLE
[Accessibility] Shift+Tab is always forced to go to a11y terminal buffer even when insert tab character mode is used

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#174800]
CC @meganrogge

## Reproducible Steps

1. Open terminal.

1. Launch `ipython`

1. Press Ctrl+M to toggle on insert tab character mode.

1. In ipython, do the following,

``` python
import p
# After p key, press tab key it cycles through the next suggestions.
# This time, try with Shift+Tab key to see if that cycles back through the list.
```
## Current Behavior

Shift+Tab always goes to the terminal a11y buffer.

## Expected Behavior

When tab key is used for inserting character mode, Shift+Tab key should be passed to terminal input.

## System Info

Version: 1.76.0-insider (system setup)
Commit: 96a795cc0b1cd62f9ff66c347637f917c6dde3da
Date: 2023-02-17T22:22:55.538Z
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Windows_NT x64 10.0.22621
Sandboxed: Yes

**Comment by meganrogge** [comment]
> Ctrl+M to toggle on insert tab character mode

I am not familiar with insert tab character mode, hopefully that's something we can detect so that it's possible to respect this

**Comment by jooyoungseo** [comment]
@meganrogge -- You know, pressing Ctrl+M key in terminal input cycles through the two different modes: "Pressing Tab will now move focus to the next focusable element" and "Pressing Tab will now insert the tab character." I was referring to the mode that says "Pressing Tab will now insert the tab character."

**Comment by meganrogge** [comment]
Oh tab focus mode - yes, we can definitely listen to that

**Comment by meganrogge** [comment]
@jooyoungseo and I discussed a better approach - remove the `Shift+Tab` keybinding, add `alt+`` agnostic to whether the terminal is focused

**Comment by jooyoungseo** [comment]
@meganrogge -- FYI, I have customized my `keybindings` like below:

``` json
[
  {
    "key": "alt+`",
    "command": "workbench.action.terminal.focusAccessibleBuffer",
    "when": "accessibilityModeEnabled && terminalHasBeenCreated || accessibilityModeEnabled && terminalProcessSupported"
  },
  {
    "key": "shift+tab",
    "command": "-workbench.action.terminal.focusAccessibleBuffer",
    "when": "accessibilityModeEnabled && terminalFocus && terminalHasBeenCreated || accessibilityModeEnabled && terminalFocus && terminalProcessSupported"
  }
]
```

Currently, I have the following two issues:

1. `alt+`` does not work when my focus is in terminal input.

1. `Alt+`` does not work when terminal is created in editor area.

**Comment by meganrogge** [comment]
@jooyoungseo in today's insider's, I cannot reproduce the issue you describe above so believe it's been fixed. to clarify, I toggle off `Tab Moves Focus Mode` in order to get the tab completion to work with ipython.

I'll open a separate issue and describe your reasoning behind switching from tab+shift

**Comment by jooyoungseo** [comment]
@meganrogge -- that's odd. This issue still exits on my end. Tested with today's insiders on Windows 11. NVDA and JAWS have the same issue. While pressing Shift+Tab works as a key to go back to the previous suggestion in an external terminal, it does not work in VSCode terminal. It always goes to the previous focus tab (i.e., terminal buffer). FYI, I have turned off the terminal tab focus mode.

**Comment by meganrogge** [comment]
I misunderstood - I was pressing only tab and I now can reproduce the issue when pressing shift+tab. I think the fix for this will be to use a different keybinding as will be tracked in https://github.com/microsoft/vscode/issues/175162. It makes sense to use something that terminals don't already respond to as shift+tab also does special things in oh-my-zsh for example.

**Comment by meganrogge** [comment]
we will only focus the accessible buffer when tab focus mode is enabled
